src/middleware/types.ts

```diff
@@ -0,0 +1,16 @@
+export type GeneralRequest = {
+  method: string;
+  url: string;
+  headers: Record<string, string>;
+  text: () => Promise<string>;
```

Comment:
i'm not 100% sure about this API surface. I guess you are trying to stay close to the web platform [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object? We can leave it as is because I don't have a better idea right now, maybe it will grow on me 🤷🏼